experiments/word2vec/arxiv_fetcher.py

```
@@ -0,0 +1,371 @@
+
+        Returns:
+            Concatenated LaTeX content or None if extraction fails
+        """
+        import tarfile
+        import tempfile
+
+        try:
+            # Extract tarball to temp directory
+            with tempfile.TemporaryDirectory() as tmpdir:
+                with tarfile.open(latex_path, 'r:gz') as tar:
+                    tar.extractall(tmpdir)
```

Ruff (0.13.3)

Uses of `tarfile.extractall()`

(S202)